lib/tree_sitter/node.rb

```diff
@@ -163,5 +163,94 @@ def fetch(*keys)
       end
       fields.values_at(*keys)
     end
+
+    # Regex for line annotatin extraction from sexpr with source.
+    #
+    # @!visibility private
+    LINE_ANNOTATION = /@\{([^}]*)\}/
+
+    # Pretty-prints the node's sexp.
+    #
+    # The default call to {to_s} or {to_string} calls tree-sitter's
+    # `ts_node_string`. It's displayed on a single line, so reading a rich node
+    # become tiresome.
+    #
+    # This provides a better sexpr where you can control the "screen" width to
+    # decide when to break.
+    #
+    # @param indent   [Integer]
+    #   indentation for nested nodes.
+    # @param width    [Integer]
+    #   the screen's width.
+    # @param source   [Nil|String]
+    #   display source on the margin if not `nil`.
+    # @param vertical [Nil|Boolean]
+    #   fit as much sexpr on a single line if `false`, else, go vertical.
+    #   This is always `true` if `source` is not `nil`.
+    #
+    # @return [String] the pretty-printed sexpr.
+    def sexpr(indent: 2, width: 120, source: nil, vertical: nil)
+      res =
+        sexpr_recur(
+          indent: indent,
+          width: width,
+          source: source,
+          vertical: (!source.nil? || !!vertical),
+        ).output
+      return res if source.nil?
+
+      max_width = 0
+      res
+        .lines
+        .map { |line|
+          extracted = line.scan(LINE_ANNOTATION).flatten.first || ''
+          base = line.gsub(LINE_ANNOTATION, '').rstrip
+          max_width = [max_width, base.length].max
+          [base, extracted]
+        }
+        .map { |base, extracted|
```

Comment:
This intermediary map seems really useless.

Reply:
I need it to compute max_width …